cpp/include/cuvs/neighbors/mg_ivf_flat.h

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2024-2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuvs/core/c_api.h>
+#include <cuvs/neighbors/ivf_flat.h>
+#include <cuvs/neighbors/mg_common.h>
+#include <dlpack/dlpack.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup mg_ivf_flat_c_index_params Multi-GPU IVF-Flat index build parameters
+ * @{
+ */
+
+/**
+ * @brief Multi-GPU parameters to build IVF-Flat Index
+ *
+ * This structure extends the base IVF-Flat index parameters with multi-GPU specific settings.
+ */
+struct cuvsMgIvfFlatIndexParams {
+  /** Base IVF-Flat index parameters */
+  cuvsIvfFlatIndexParams_t base_params;
+  /** Distribution mode for multi-GPU setup */
+  cuvsMgDistributionMode mode;
+};
+
+typedef struct cuvsMgIvfFlatIndexParams* cuvsMgIvfFlatIndexParams_t;
+
+/**
+ * @brief Allocate Multi-GPU IVF-Flat Index params, and populate with default values
+ *
+ * @param[in] index_params cuvsMgIvfFlatIndexParams_t to allocate
+ * @return cuvsError_t
+ */
+cuvsError_t cuvsMgIvfFlatIndexParamsCreate(cuvsMgIvfFlatIndexParams_t* index_params);
+
+/**
+ * @brief De-allocate Multi-GPU IVF-Flat Index params
+ *
+ * @param[in] index_params
+ * @return cuvsError_t
+ */
+cuvsError_t cuvsMgIvfFlatIndexParamsDestroy(cuvsMgIvfFlatIndexParams_t index_params);
```

Comment:
As mentioned above, I know it's verbose, but `MultiGpu` would be much easier to spot